src/flamenco/runtime/program/vote/fd_vote_state_versioned.c

```diff
@@ -0,0 +1,762 @@
+      return fd_vote_state_v4_set_new_authorized_voter(
+          ctx,
+          &self->inner.v4,
+          authorized_pubkey,
+          current_epoch,
+          target_epoch,
+          authorized_withdrawer_signer,
+          signers
+      );
+    default:
+      FD_LOG_CRIT(( "unsupported vote state version: %u", self->discriminant ));
+  }
+}
+
+void
+fd_vsv_set_node_pubkey( fd_vote_state_versioned_t * self,
+                        fd_pubkey_t const *         node_pubkey ) {
+  switch( self->discriminant ) {
+    case fd_vote_state_versioned_enum_v3:
+      self->inner.v3.node_pubkey = *node_pubkey;
+      break;
+    case fd_vote_state_versioned_enum_v4:
+      self->inner.v4.node_pubkey = *node_pubkey;
+      break;
+  }
+}
+
+void
+fd_vsv_set_block_revenue_collector( fd_vote_state_versioned_t * self,
+                                    fd_pubkey_t const *         block_revenue_collector ) {
+  switch( self->discriminant ) {
+    case fd_vote_state_versioned_enum_v4:
+      self->inner.v4.block_revenue_collector = *block_revenue_collector;
+      break;
+  }
+}
+
+void
+fd_vsv_set_commission( fd_vote_state_versioned_t * self,
+                       uchar                       commission ) {
+  switch( self->discriminant ) {
+    case fd_vote_state_versioned_enum_v3:
+      self->inner.v3.commission = commission;
+      break;
+    case fd_vote_state_versioned_enum_v4:
+      self->inner.v4.inflation_rewards_commission_bps = commission*100;
+      break;
+    default:
+      FD_LOG_ERR(( "unsupported vote state version: %u", self->discriminant ));
+  }
+}
+
+void
+fd_vsv_set_root_slot( fd_vote_state_versioned_t * self, ulong * root_slot ) {
+  switch( self->discriminant ) {
+    case fd_vote_state_versioned_enum_v3:
+      self->inner.v3.has_root_slot = (root_slot!=NULL);
+      if( FD_LIKELY( root_slot ) ) {
+        self->inner.v3.root_slot = *root_slot;
+      }
+      break;
+    case fd_vote_state_versioned_enum_v4:
+      self->inner.v4.has_root_slot = (root_slot!=NULL);
+      if( FD_LIKELY( root_slot ) ) {
+        self->inner.v4.root_slot = *root_slot;
+      }
+      break;
+  }
+}
+
+static void
+fd_vsv_set_last_timestamp( fd_vote_state_versioned_t *       self,
+                           fd_vote_block_timestamp_t const * last_timestamp ) {
+  switch( self->discriminant ) {
+    case fd_vote_state_versioned_enum_v3:
+      self->inner.v3.last_timestamp = *last_timestamp;
+      break;
+    case fd_vote_state_versioned_enum_v4:
+      self->inner.v4.last_timestamp = *last_timestamp;
+      break;
+    default:
+      FD_LOG_ERR(( "unsupported vote state version: %u", self->discriminant ));
+  }
+}
+
+/**********************************************************************/
+/* General functions                                                  */
+/**********************************************************************/
+
+// https://github.com/anza-xyz/agave/blob/v2.0.1/sdk/program/src/vote/state/mod.rs#L855
```

Comment:
Use `/* */` comment format to avoid possible breakage of linting tools

(Unless this is just copy pasted)